File: Client/src/client/systems/ped.js
import alt from 'alt-client';
import native from 'natives';


function createPed(type, hash, positionX, positionY, positionZ, rotation) {
    const modelHash = alt.hash(hash);
    alt.loadModel(modelHash);
    const ped = native.createPed(type, modelHash, positionX, positionY, positionZ, rotation, false, false);
    native.freezeEntityPosition(ped, true);
    native.setEntityInvincible(ped, true);
    native.setBlockingOfNonTemporaryEvents(ped, true);
    native.setEntityRotation(ped, 0, 0, rotation, 2, true);
    const test = native.getEntityRotation(ped, 2);
    alt.log(test);
}


alt.onServer('Client:Ped:Create', (type, hash, positionX, positionY, positionZ, rotation) => {
    createPed(type, hash, positionX, positionY, positionZ, rotation, false, false);
});

File: Los Angeles Life Client/Los Angeles Life Client/System/Ped.cs
using AltV.Net.Client;

namespace Los_Angeles_Life_Client.System;

public abstract class Ped : Client
{
    public static void Load()
    {
        Alt.OnServer("Client:Ped:Create", (int type, string hash, float positionX, float positionY, float positionZ, float rotation, bool isNewPed) =>
        {
            CreatePed(type, hash, positionX, positionY, positionZ, rotation, isNewPed);
        });
    }

    private static void CreatePed(int type, string hash, float positionX, float positionY, float positionZ, float rotation, bool isNewPed)
    {
        if(!isNewPed) return;

        var modelHash = Alt.Hash(hash);
        Alt.LoadModel(modelHash);
        var ped = Alt.Natives.CreatePed(type, modelHash, positionX, positionY, positionZ, rotation, false, false);
        Alt.Natives.FreezeEntityPosition(ped, true);
        Alt.Natives.SetEntityInvincible(ped, true);
        Alt.Natives.SetBlockingOfNonTemporaryEvents(ped, true);
        Alt.Natives.SetEntityRotation(ped, 0, 0, rotation, 2, true);
    }
}